src/compas_rhino/geometry/surface.py

```diff
@@ -430,7 +430,7 @@ def mesh_uv_to_xyz(self, mesh, cls=None):
             cls = type(mesh)
 
         vertices, faces = mesh.to_vertices_and_faces()
-        vertices = [self.point_uv_to_xyz(uv0[:2]) for uv0 in vertices]
+        vertices = {vkey: self.point_uv_to_xyz(uv0[:2]) for vkey, uv0 in vertices.items()}
```

Comment:
what is `uv0`? (or why the zero)

Reply:
because I output a 3D point (u, v, 0) and not a 2D point (u, v)

Reply:
so uv0[:2] is (u, v)

Reply:
true it's misleading, I changed to 2D inputs and outputs for uv
the user will do the formatting